### DISCUSSION TITLE
How do I remove the SortIcon space from DataGrid header?

### DISCUSSION THREAD

**Discussion by alessandrofrancesconi** [AvaloniaUI/Avalonia#10544] [Q&A] [answered]
My DataGrid is defined as:

```
<DataGrid Name="GridRecords" Grid.Row="1" ColumnWidth="*"
			IsReadOnly="True"
			SelectionMode="Single"
			RowBackground="Black" GridLinesVisibility="All"
			HeadersVisibility="Column" 
			CanUserSortColumns="False" CanUserReorderColumns="False" CanUserResizeColumns="False"
			Items="{Binding RecordsCollection}">
	<DataGrid.Columns>
		<DataGridTemplateColumn>
			<DataGridTemplateColumn.Header>
				<ComboBox Name="ComboCompareDriver1" Items="{Binding ComparableDrivers}" HorizontalAlignment="Stretch" SelectionChanged="ComboCompareDriver_SelectionChanged" />
			</DataGridTemplateColumn.Header>
			<DataGridTemplateColumn.CellTemplate>
				<DataTemplate>
					<TextBlock Text="{Binding Path=TimeCompareDriver1, Mode=OneTime}" VerticalAlignment="Center" Margin="5,0" />
				</DataTemplate>
			</DataGridTemplateColumn.CellTemplate>
		</DataGridTemplateColumn>

		<DataGridTemplateColumn>
			<DataGridTemplateColumn.Header>
				<ComboBox Name="ComboCompareDriver2" Items="{Binding ComparableDrivers}" HorizontalAlignment="Stretch" SelectionChanged="ComboCompareDriver_SelectionChanged" />
			</DataGridTemplateColumn.Header>
			<DataGridTemplateColumn.CellTemplate>
				<DataTemplate>
					<TextBlock Text="{Binding Path=TimeCompareDriver2, Mode=OneTime}" VerticalAlignment="Center" Margin="5,0" />
				</DataTemplate>
			</DataGridTemplateColumn.CellTemplate>
		</DataGridTemplateColumn>
	</DataGrid.Columns>
</DataGrid>
```

I see it outputs a space right after each ComboBox, in the header:

![image](https://user-images.githubusercontent.com/1595223/222751920-6b513206-e1c8-47cc-982b-9254f6fd041a.png)

By looking at the visual tree, I suppose they are spaces for "SortIcon", which I do not need. How can I remove them and gain space for the ComboBoxes?

**Comment by timunie** [comment, marked as answer]
Look up the original Style. I think you need to override the Control template. Assuming you are using Avalonia 0.10.18, it is here: 

https://github.com/AvaloniaUI/Avalonia/blob/401bc521d2947bf7699a530f3d6e43e7a32c6baf/src/Avalonia.Controls.DataGrid/Themes/Fluent.xaml#L186-L240

This is the problematic line (Remove the MinWidth): 

https://github.com/AvaloniaUI/Avalonia/blob/401bc521d2947bf7699a530f3d6e43e7a32c6baf/src/Avalonia.Controls.DataGrid/Themes/Fluent.xaml#L197

**Reply by alessandrofrancesconi** [reply]
Thanks!

**Comment by mythter** [comment]
I came up to this solution:
when no sorting is applied the sort icon is not visible so you can just span the header name column to sort icon column in internal `Grid`, and when sorting is used the icon will appear as usual
```
<DataGrid.Styles>
	<Style Selector="DataGridColumnHeader:not(:sortascending):not(:sortdescending) ContentPresenter#PART_ContentPresenter">
		<Setter Property="Grid.ColumnSpan" Value="2"/>
	</Style>
</DataGrid.Styles>
```
